### PR TITLE
Ndlano/issues#1057 related external articles

### PR DESCRIPTION
**Pull request by sebastianjg**
resolves https://github.com/NDLANO/Issues/issues/1057 -> See issue for embed examples.
needs https://github.com/NDLANO/frontend-packages/pull/231 - Note: test will fail until `ndla-ui`is updated.

### CODE REVIEW

File: src/plugins/relatedContentPlugin.js [hunk 1]
```diff
@@ -22,109 +20,159 @@ const RESOURCE_TYPE_SUBJECT_MATERIAL = 'urn:resourcetype:subjectMaterial';
...
+    return {
+      ...embed,
+      article: article ? { ...article, resource } : undefined,
+    };
   }
 
-  const embedToHTML = (embed, locale) => {
-    if (!embed.articles || embed.articles.length === 0) {
+  const embedToHTML = embed => {
+    if (!embed.article) {
```
[the remark this answers is not in the document]
Comment (sebastianjg) [reply src/plugins/relatedContentPlugin.js]: `article` settes til `undefined` dersom den ikke klarer å fetche disse.

File: src/plugins/relatedContentPlugin.js [hunk 2]
```diff
@@ -22,109 +20,159 @@ const RESOURCE_TYPE_SUBJECT_MATERIAL = 'urn:resourcetype:subjectMaterial';
...
+    if (embed.data.url) {
+      article = {
+        title: { title: embed.data.title },
+        metaDescription: {
+          metaDescription: embed.data.metaDescription || embed.data.url,
+        },
+        linkInfo: `${t(lang, 'related.linkInfo')} ${
+          embed.data.url.match(
+            /^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:/\n]+)/im
+          )[1]
```
[the remark this answers is not in the document]
Comment (sebastianjg) [reply src/plugins/relatedContentPlugin.js]: Ikke helt så rett frem med server side node 😃

File: src/plugins/relatedContentPlugin.js [hunk 3]
```diff
@@ -22,109 +22,151 @@ const RESOURCE_TYPE_SUBJECT_MATERIAL = 'urn:resourcetype:subjectMaterial';
...
-    );
+    if (embed.data.url) {
+      article = {
+        title: { title: embed.data.title },
+        metaDescription: {
+          metaDescription: embed.data.metaDescription || embed.data.url,
+        },
+        linkInfo: `${t(lang, 'related.linkInfo')} ${
+          // Get domain name only from url
+          embed.data.url.match(
```
Comment (oyvinmar) [inline src/plugins/relatedContentPlugin.js]: Should probably fallback to embed.data.url if match fails (i.e. undefined).

File: src/__tests__/integration-tests/article417/article417.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ export default {
...
...keren og for å sikre at riktig informasjon blir utfylt.</p><p>For å sjekke et telefonnummer trengs et skript som sjekker:</p><ul><li>om det kun er skrevet inn tall</li><li>om det er åtte siffer</li></ul><p>Skriptet kan da gi en feilmelding om «Feilskrevet telefonnummer» eller en mer detaljert feilmelding som «Du har skrevet tegn som ikke er tall». Sannsynligvis burde skriptet være så komplekst at det tok høyde for valgfri landskode (f.eks. +47).</p><p>Vi kan bruke valg-setninger for å teste:</p><pre><code><strong>if</strong> (telefonFelt.value.length != 8) {    /* != betyr ulik */<br>         alert("Du må skrive inn åtte tall.");     <br> } </code></pre><p>Bruk av <strong>valg</strong> er en av de store styrkene til et programmeringsspråk, noe du fort vil erfare. Programmering av <strong>gjentakelser</strong> med løkker er en annen. Vi kan sette opp en løkke som for eksempel repeterer samme testkode for alle felt i skjemaet vårt.</p></section><section><div data-type="related-content"><embed data-article-id="413" data-resource="related-content"><embed data-article-id="414" data-resource="related-content"><embed data-article-id="416" data-resource="related-content"></div></section>',
```
Comment (oyvinmar) [inline src/__tests__/integration-tests/article417/article417.js]: Maybe add a integration test for external related article to.

**Pull request merged**